docs/API.md

```diff
@@ -437,8 +411,82 @@ Options
 There's a general method of setting options on a socket in the SRT C API, similar
 to the system setsockopt/getsockopt functions.
 
-Synopsis
---------
+Types used in socket options
+----------------------------
+
+Possible types of options are:
+
+* `int32_t` - usually you can treat it as `int` equivalent, as this type
+wasn't changed the size on 64-bit systems, although the definition is using
+`int32_t` type so that the bit size is fixed by definition. In some cases
+the value is expressed using an enumeration type (see below).
+
+* `int64_t` - some options need the paramter specified as 64-bit integer
+
+* `bool` - You need to use a boolean type, the builtin one in C++ and the
+one from `<stdbool.h>` for C, although when setting an option, passing the
+value through `int` type is also tolerated and properly recognized. When
+getting an option, however, you better use the `bool` type, although you
+can risk passing a variable of `int` type initialized with 0 and then check if
+the resulting value is equal to 0 (just don't compare the result with 1).
+
+* `string` - When setting, you pass the character array pointer as value
+and the string length as length. When getting, you have to pass a long
+enough array (with specified size in the size variable). Every option with
+this type that can be read should specify the maximum length of that array.
+
+
+Enumeration types used in options
+---------------------------------
+
+
+### 1. `SRT_TRANSTYPE`. Used by `SRTO_TRANSTYPE` option:
+
+* `SRTT_LIVE`: Live mode.
+* `SRTT_FILE`: File mode.
+
+See below [Transmission types](#transmission-types) for details.
+
+
+### 2. `SRT_KM_STATE`. This defined encryption state as performed by the Key
+Material Exchange, used by `SRTO_RCVKMSTATE`, `SRTO_SNDKMSTATE` and
+`SRTO_KMSTATE` options:
+
+- `SRT_KM_S_UNSECURED`: no encryption/descryption. If this state is only on
+the receiver, received encrypted packets will be dropped.
+
+- `SRT_KM_S_SECURING`: pending security (HSv4 only). This is a temporary state
+used only if the connection is made has HSv4 and the Key Material Exchange is
+not finished yet. On HSv5 not possible because Key Material Exchange for the
+initial key is done in the handshake.
+
+- `SRT_KM_S_SECURED`: KM exchange was successful and the data will be sent
+enrypted and decrypted by the receiver. This state is only possible on both
+sides in both directions simultaneously.
+
+- `SRT_KM_S_NOSECRET`: (HSv5 only) This site has set password, but data will 
+be received as plain. This also means that sending the data from this party
+will not be received by the peer due to impossible decryption.
```

Comment:
@ethouris This description is unclear to me, but here is my attempt at rewording it:

```suggestion
- `SRT_KM_S_NOSECRET`: (HSv5 only) The sender has set a password, but data will 
be received as plain. This also means it will be impossible for the receiver to decrypt any data being sent.
```

Reply:
I said "this site" because I meant the site where this state is visible.

To make you imagine the situation: this is when you have two parties:

[A] -> password=(something was set), state=SRT_KM_S_NOSECRET
[B] -> password=(not set), state=SRT_KM_S_UNSECURED

In this configuration [B] will be able to send payloads to [A], unencrypted, and [A] will correctly receive it, while although [A] could send packets to [B], but they will be encrypted and [B] will be unable to decrypt them.

The description is for what exactly the state is on the [A] site. I'm not changing it now, proposed description would be:

This state means that the agent has set the password, but the peer did not. In this case the agent can receive unencrypted packets from the peer, but packets sent by the agent to the peer will be encrypted and the peer will not be able to decrypt them.

Reply:
How about:

`If this state is in the sending direction (`SRTO_SNDKMSTATE`), then it means that the sending party 
has set a passphrase, but the peer did not. In this case the sending party can receive unencrypted packets 
from the peer, but packets it sends to the peer will be encrypted and the peer will not be able to decrypt them. 
This state is only possible in HSv5.
`

Reply:
Perfect.